yarn-project/types/src/tx.ts

```diff
@@ -194,6 +194,33 @@ export class Tx {
   static async getHashes(txs: Tx[]): Promise<TxHash[]> {
     return await Promise.all(txs.map(tx => tx.getTxHash()));
   }
+
+  /**
+   * Clones a tx, making a deep copy of all fields.
+   * @param tx - The transaction to be cloned.
+   * @returns The cloned transaction.
+   */
+  static clone(tx: Tx): Tx {
```

Comment:
I think [`cloneDeep`](https://www.npmjs.com/package/lodash.clonedeep) could handle this automatically